public/app/plugins/datasource/tempo/_importedDependencies/datasource_srv.ts

```diff
@@ -0,0 +1,10 @@
+import { DataSourceInstanceSettings } from '@grafana/data';
+import { getDataSourceSrv as getDataSourceService, DataSourceSrv as DataSourceService } from '@grafana/runtime';
+
+export const getDatasourceSrv = (): DatasourceSrv => {
+  return getDataSourceService() as DatasourceSrv;
+};
+
+export interface DatasourceSrv extends DataSourceService {
+  getDataSourceSettingsByUid(uid: string): DataSourceInstanceSettings | undefined;
```

Comment:
this won't work? You are using the datasource srv from @grafana/runtime which won't have this method. I think you can just omit this and change Tempo to use `getInstanceSettings` which apparently already accepts a UID as the parameter ([example](https://github.com/grafana/grafana/blob/main/public/app/features/alerting/unified/components/rule-editor/QueryRows.tsx#L143)).